File: strings/practice questions/Mars_exploration_hari.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the marsExploration function below.
def marsExploration(s):
    ref="SOS"
    count=0
    for i in range(len(s)):
        if(ref[i%3]!=s[i]):
            count+=1
    return(count)


if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    s = input()

    result = marsExploration(s)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Hackerrank/strings/practice questions/Mars_exploration_hari.cpp
#include <bits/stdc++.h>

using namespace std;

// Complete the marsExploration function below.
int marsExploration(string s) {
    string ref="SOS";
    int count=0;
    int i,l;
    l=s.length();
    for(i=0;i<l;i++)
    {
        if(ref[i%3]!=s[i])
        {
            count++;
        }
    }
    return(count);


}

int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string s;
    getline(cin, s);

    int result = marsExploration(s);

    fout << result << "\n";

    fout.close();

    return 0;
}
